Consider bumping the version of dependency scrypt.js from 0.2.0 to 0.3.0

#### Expected behavior

 The now deprecated [node-scrypt](https://github.com/barrysteyn/node-scrypt) should be an optional dependency.

#### Actual behavior

web3-eth-accounts depends on [scrypt.js](https://github.com/axic/scrypt.js) which has recently released a new [version.](https://github.com/axic/scrypt.js/pull/3)

This version allows the users to optionally exclude the package [node-scrypt](https://github.com/barrysteyn/node-scrypt)

node-scrypt is treated as an optional dependency and users specifying --no-optional will install the JS-version of scrypt, scryptsy, only. 

This is important because node-scrypt relies on node-gyp bindings and is problematic to install on some systems. 

By bumping scrypt.js to 0.3.0 we will also be making node-scrypt optional.

Users not specifying --no-optional on installation will not be affected by this change. 

#### Steps to reproduce the behavior
1. npm install --save web3@1.0.0-beta.46
2. it installs scrypt.js 0.2.0

#### Versions
web3.js 1.0

On a side note, node-scrypt is considered deprecated by the author since Node now has a standard library for [scrypt.](https://nodejs.org/api/crypto.html#crypto_crypto_scrypt_password_salt_keylen_options_callback)

It would be awesome if we could move to this long-term.

Hi there,

**Context**
I'm facing a problem with one of my side-project.
I really want to put a step on blockchain development, and I'm trying to build [my first front fetching accounts from blockchain](https://github.com/Ugarz/eth-todo-list-react). I went to [Spectrum chat](https://spectrum.chat/next-js/general/error-while-building-project~6d28afb8-36b7-4ec3-a09c-902afe2463e2) to see if it was a Nextjs related issue, someone in comments found that the `scrypt` module were required by a `web3` dependency called `ethers` [(cf: ethers npm package)](https://github.com/ethers-io/ethers.js).

I am following a step by step introduction [from this video (made by Gregory)](https://youtu.be/AiWkkj8lSTc?t=412) where he is using the `1.0.0-beta.46`. I'm currently using the `1.0.0-beta.48`, but I still get the error : 
```
Module not found: Can't resolve './build/Release/scrypt' in '/tmp/13f82355/node_modules/scrypt'
```

**Here my questions** 
- Can I now use correctly web3 without my `scrypt error` with a specific release of web3 ?
- Does using `crypto.scrypt` module (cf: [Node Crypto Scrypt Module](https://nodejs.org/api/crypto.html#crypto_crypto_scrypt_password_salt_keylen_options_callback)) would solve this as @Kaisle suggests it ?

--- a/packages/web3-eth-accounts/package.json
+++ b/packages/web3-eth-accounts/package.json
@@ -17,10 +17,9 @@
     },
     "dependencies": {
         "@babel/runtime": "^7.3.1",
-        "crypto-browserify": "3.12.0",
         "eth-lib": "0.2.8",
         "lodash": "^4.17.11",
-        "scrypt.js": "0.2.0",
+        "scrypt.js": "0.3.0",
         "uuid": "3.3.2",
         "web3-core": "1.0.0-beta.54",
         "web3-core-helpers": "1.0.0-beta.54",

--- a/packages/web3-eth-accounts/package.json
+++ b/packages/web3-eth-accounts/package.json
@@ -17,6 +17,9 @@
     },
     "dependencies": {
         "@babel/runtime": "^7.3.1",
+        "randombytes": "^2.1.0",
+        "browserify-cipher": "^1.0.1",
+        "pbkdf2": "^3.0.17",
         "eth-lib": "0.2.8",
         "lodash": "^4.17.11",
         "scrypt.js": "0.3.0",

--- a/packages/web3-eth-accounts/src/models/Account.js
+++ b/packages/web3-eth-accounts/src/models/Account.js
@@ -23,8 +23,10 @@ import isObject from 'lodash/isObject';
 import * as EthLibAccount from 'eth-lib/lib/account'; // TODO: Remove this dependency
 import uuid from 'uuid';
 import Hash from 'eth-lib/lib/hash';
+import randomBytes from 'randombytes';
+import {pbkdf2Sync} from 'pbkdf2';
+import {createCipheriv, createDecipheriv} from 'browserify-cipher';
 import {isHexStrict, hexToBytes, randomHex, keccak256} from 'web3-utils'; // TODO: Use the VO's of a web3-types module.
-import crypto from 'crypto';
 
 export default class Account {
     /**
@@ -135,8 +137,8 @@ export default class Account {
      */
     toV3Keystore(password, options) {
         options = options || {};
-        const salt = options.salt || crypto.randomBytes(32);
-        const iv = options.iv || crypto.randomBytes(16);
+        const salt = options.salt || randomBytes(32);
+        const iv = options.iv || randomBytes(16);
 
         let derivedKey;
         const kdf = options.kdf || 'scrypt';
@@ -148,7 +150,7 @@ export default class Account {
         if (kdf === 'pbkdf2') {
             kdfparams.c = options.c || 262144;
             kdfparams.prf = 'hmac-sha256';
-            derivedKey = crypto.pbkdf2Sync(Buffer.from(password), salt, kdfparams.c, kdfparams.dklen, 'sha256');
+            derivedKey = pbkdf2Sync(Buffer.from(password), salt, kdfparams.c, kdfparams.dklen, 'sha256');
         } else if (kdf === 'scrypt') {
             // FIXME: support progress reporting callback
             kdfparams.n = options.n || 8192; // 2048 4096 8192 16384
@@ -159,7 +161,7 @@ export default class Account {
             throw new Error('Unsupported kdf');
         }
 
-        const cipher = crypto.createCipheriv(options.cipher || 'aes-128-ctr', derivedKey.slice(0, 16), iv);
+        const cipher = createCipheriv(options.cipher || 'aes-128-ctr', derivedKey.slice(0, 16), iv);
         if (!cipher) {
             throw new Error('Unsupported cipher');
         }
@@ -176,7 +178,7 @@ export default class Account {
 
         return {
             version: 3,
-            id: uuid.v4({random: options.uuid || crypto.randomBytes(16)}),
+            id: uuid.v4({random: options.uuid || randomBytes(16)}),
             address: this.address.toLowerCase().replace('0x', ''),
             crypto: {
                 ciphertext: ciphertext.toString('hex'),
@@ -239,7 +241,7 @@ export default class Account {
                 throw new Error('Unsupported parameters to PBKDF2');
             }
 
-            derivedKey = crypto.pbkdf2Sync(
+            derivedKey = pbkdf2Sync(
                 Buffer.from(password),
                 Buffer.from(kdfparams.salt, 'hex'),
                 kdfparams.c,
@@ -257,7 +259,7 @@ export default class Account {
             throw new Error('Key derivation failed - possibly wrong password');
         }
 
-        const decipher = crypto.createDecipheriv(
+        const decipher = createDecipheriv(
             json.crypto.cipher,
             derivedKey.slice(0, 16),
             Buffer.from(json.crypto.cipherparams.iv, 'hex')

--- a/packages/web3-eth-accounts/src/models/Account.js
+++ b/packages/web3-eth-accounts/src/models/Account.js
@@ -24,7 +24,7 @@ import * as EthLibAccount from 'eth-lib/lib/account'; // TODO: Remove this depen
 import uuid from 'uuid';
 import Hash from 'eth-lib/lib/hash';
 import {isHexStrict, hexToBytes, randomHex, keccak256} from 'web3-utils'; // TODO: Use the VO's of a web3-types module.
-const crypto = typeof global === 'undefined' ? require('crypto-browserify') : require('crypto');
+import crypto from 'crypto';
 
 export default class Account {
     /**

--- a/packages/web3-eth-accounts/tests/src/models/AccountTest.js
+++ b/packages/web3-eth-accounts/tests/src/models/AccountTest.js
@@ -1,5 +1,7 @@
 import scryptsy from 'scrypt.js';
-import crypto from 'crypto';
+import randomBytes from 'randombytes';
+import {pbkdf2Sync} from 'pbkdf2';
+import {createCipheriv, createDecipheriv} from 'browserify-cipher';
 import uuid from 'uuid';
 import Hash from 'eth-lib/lib/hash';
 import {fromPrivate, sign, decodeSignature} from 'eth-lib/lib/account';
@@ -13,7 +15,9 @@ jest.mock('eth-lib/lib/account');
 jest.mock('eth-lib/lib/hash');
 jest.mock('web3-utils');
 jest.mock('uuid');
-jest.mock('crypto');
+jest.mock('randombytes');
+jest.mock('pbkdf2');
+jest.mock('browserify-cipher');
 jest.mock('scrypt.js');
 jest.mock('../../../src/Accounts');
 
@@ -152,15 +156,7 @@ describe('AccountTest', () => {
 
         decipher.final.mockReturnValueOnce(Buffer.from('0'));
 
-        crypto.createDecipheriv = jest.fn((cipher, derivedKey, buffer) => {
-            expect(cipher).toEqual('cipher');
-
-            expect(derivedKey).toEqual(Buffer.from('0000000000000000'));
-
-            expect(buffer).toEqual(Buffer.from(['0x0'], 'hex'));
-
-            return decipher;
-        });
+        createDecipheriv.mockReturnValueOnce(decipher);
 
         expect(Account.fromV3Keystore(json, 'password', false)).toBeInstanceOf(Account);
 
@@ -181,7 +177,7 @@ describe('AccountTest', () => {
             Buffer.concat([Buffer.from('0000000000000000'), Buffer.from(json.crypto.ciphertext, 'hex')])
         );
 
-        expect(crypto.createDecipheriv).toHaveBeenCalled();
+        expect(createDecipheriv).toHaveBeenCalledWith('cipher', Buffer.from('0000000000000000'), Buffer.from(['0x0'], 'hex'));
 
         expect(decipher.update).toHaveBeenCalledWith(Buffer.from(json.crypto.ciphertext, 'hex'));
 
@@ -224,29 +220,9 @@ describe('AccountTest', () => {
 
         decipher.final.mockReturnValueOnce(Buffer.from('0'));
 
-        crypto.createDecipheriv = jest.fn((cipher, derivedKey, buffer) => {
-            expect(cipher).toEqual('cipher');
+        createDecipheriv.mockReturnValueOnce(decipher);
 
-            expect(derivedKey).toEqual(Buffer.from('0000000000000000'));
-
-            expect(buffer).toEqual(Buffer.from(['0x0'], 'hex'));
-
-            return decipher;
-        });
-
-        crypto.pbkdf2Sync = jest.fn((password, salt, c, dklen, sha256) => {
-            expect(password).toEqual(Buffer.from(password));
-
-            expect(salt).toEqual(Buffer.from('salt', 'hex'));
-
-            expect(c).toEqual(1);
-
-            expect(dklen).toEqual('dklen');
-
-            expect(sha256).toEqual('sha256');
-
-            return Buffer.from('00000000000000000000000000000000');
-        });
+        pbkdf2Sync.mockReturnValueOnce(Buffer.from('00000000000000000000000000000000'));
 
         expect(Account.fromV3Keystore(json, 'password', false)).toBeInstanceOf(Account);
 
@@ -254,13 +230,13 @@ describe('AccountTest', () => {
             `0x${Buffer.concat([Buffer.from('0'), Buffer.from('0')]).toString('hex')}`
         );
 
-        expect(crypto.pbkdf2Sync).toHaveBeenCalled();
+        expect(pbkdf2Sync).toHaveBeenCalledWith(Buffer.from('password'), Buffer.from('salt', 'hex'), 1, 'dklen', 'sha256');
 
         expect(keccak256).toHaveBeenCalledWith(
             Buffer.concat([Buffer.from('0000000000000000'), Buffer.from(json.crypto.ciphertext, 'hex')])
         );
 
-        expect(crypto.createDecipheriv).toHaveBeenCalled();
+        expect(createDecipheriv).toHaveBeenCalledWith('cipher', Buffer.from('0000000000000000'), Buffer.from(['0x0'], 'hex'));
 
         expect(decipher.update).toHaveBeenCalledWith(Buffer.from(json.crypto.ciphertext, 'hex'));
 
@@ -313,25 +289,13 @@ describe('AccountTest', () => {
 
         keccak256.mockReturnValueOnce('0xmac');
 
-        crypto.pbkdf2Sync = jest.fn((password, salt, c, dklen, sha256) => {
-            expect(password).toEqual(Buffer.from(password));
-
-            expect(salt).toEqual(Buffer.from('salt', 'hex'));
-
-            expect(c).toEqual(1);
-
-            expect(dklen).toEqual('dklen');
-
-            expect(sha256).toEqual('sha256');
-
-            return Buffer.from('00000000000000000000000000000000');
-        });
+        pbkdf2Sync.mockReturnValueOnce(Buffer.from('00000000000000000000000000000000'));
 
         expect(() => {
             Account.fromV3Keystore(json, 'password', false);
         }).toThrow('Key derivation failed - possibly wrong password');
 
-        expect(crypto.pbkdf2Sync).toHaveBeenCalled();
+        expect(pbkdf2Sync).toHaveBeenCalledWith(Buffer.from('password'), Buffer.from('salt', 'hex'), 1, 'dklen', 'sha256');
 
         expect(keccak256).toHaveBeenCalledWith(
             Buffer.concat([Buffer.from('0000000000000000'), Buffer.from(json.crypto.ciphertext, 'hex')])
@@ -346,7 +310,7 @@ describe('AccountTest', () => {
             address: '0xA'
         });
 
-        crypto.randomBytes.mockReturnValue(Buffer.from('random'));
+        randomBytes.mockReturnValue(Buffer.from('random'));
 
         const cipher = {
             update: jest.fn(),
@@ -357,7 +321,7 @@ describe('AccountTest', () => {
 
         cipher.final.mockReturnValueOnce(Buffer.from('0'));
 
-        crypto.createCipheriv.mockReturnValue(cipher);
+        createCipheriv.mockReturnValue(cipher);
 
         scryptsy.mockReturnValueOnce(Buffer.from('0000000000000000'));
 
@@ -387,15 +351,15 @@ describe('AccountTest', () => {
 
         expect(fromPrivate).toHaveBeenCalledWith('pk');
 
-        expect(crypto.randomBytes).toHaveBeenNthCalledWith(1, 32);
+        expect(randomBytes).toHaveBeenNthCalledWith(1, 32);
 
-        expect(crypto.randomBytes).toHaveBeenNthCalledWith(2, 16);
+        expect(randomBytes).toHaveBeenNthCalledWith(2, 16);
 
-        expect(crypto.randomBytes).toHaveBeenNthCalledWith(3, 16);
+        expect(randomBytes).toHaveBeenNthCalledWith(3, 16);
 
         expect(scryptsy).toHaveBeenCalledWith(Buffer.from('password'), Buffer.from('random'), 8192, 8, 1, 32);
 
-        expect(crypto.createCipheriv).toHaveBeenCalledWith(
+        expect(createCipheriv).toHaveBeenCalledWith(
             'aes-128-ctr',
             Buffer.from('0000000000000000').slice(0, 16),
             Buffer.from('random')
@@ -423,7 +387,7 @@ describe('AccountTest', () => {
             address: '0xA'
         });
 
-        crypto.randomBytes.mockReturnValue(Buffer.from('random'));
+        randomBytes.mockReturnValue(Buffer.from('random'));
 
         const cipher = {
             update: jest.fn(),
@@ -434,11 +398,9 @@ describe('AccountTest', () => {
 
         cipher.final.mockReturnValueOnce(Buffer.from('0'));
 
-        crypto.createCipheriv.mockReturnValue(cipher);
+        createCipheriv.mockReturnValue(cipher);
 
-        crypto.pbkdf2Sync = jest.fn(() => {
-            return Buffer.from('0000000000000000');
-        });
+        pbkdf2Sync.mockReturnValueOnce(Buffer.from('0000000000000000'));
 
         keccak256.mockReturnValueOnce('0xmac');
 
@@ -465,13 +427,13 @@ describe('AccountTest', () => {
 
         expect(fromPrivate).toHaveBeenCalledWith('pk');
 
-        expect(crypto.randomBytes).toHaveBeenNthCalledWith(1, 32);
+        expect(randomBytes).toHaveBeenNthCalledWith(1, 32);
 
-        expect(crypto.randomBytes).toHaveBeenNthCalledWith(2, 16);
+        expect(randomBytes).toHaveBeenNthCalledWith(2, 16);
 
-        expect(crypto.randomBytes).toHaveBeenNthCalledWith(3, 16);
+        expect(randomBytes).toHaveBeenNthCalledWith(3, 16);
 
-        expect(crypto.pbkdf2Sync).toHaveBeenCalledWith(
+        expect(pbkdf2Sync).toHaveBeenCalledWith(
             Buffer.from('password'),
             Buffer.from('random'),
             262144,
@@ -479,7 +441,7 @@ describe('AccountTest', () => {
             'sha256'
         );
 
-        expect(crypto.createCipheriv).toHaveBeenCalledWith(
+        expect(createCipheriv).toHaveBeenCalledWith(
             'aes-128-ctr',
             Buffer.from('0000000000000000').slice(0, 16),
             Buffer.from('random')
@@ -505,7 +467,7 @@ describe('AccountTest', () => {
             address: '0xA'
         });
 
-        crypto.randomBytes.mockReturnValue(Buffer.from('random'));
+        randomBytes.mockReturnValue(Buffer.from('random'));
 
         expect(() => {
             Account.fromPrivateKey('pk').toV3Keystore('password', {kdf: 'nope'});
@@ -513,9 +475,9 @@ describe('AccountTest', () => {
 
         expect(fromPrivate).toHaveBeenCalledWith('pk');
 
-        expect(crypto.randomBytes).toHaveBeenNthCalledWith(1, 32);
+        expect(randomBytes).toHaveBeenNthCalledWith(1, 32);
 
-        expect(crypto.randomBytes).toHaveBeenNthCalledWith(2, 16);
+        expect(randomBytes).toHaveBeenNthCalledWith(2, 16);
     });
 
     it('calls encrypt with a unsupported cipher', () => {
@@ -526,13 +488,11 @@ describe('AccountTest', () => {
             address: '0xA'
         });
 
-        crypto.randomBytes.mockReturnValue(Buffer.from('random'));
+        randomBytes.mockReturnValue(Buffer.from('random'));
 
-        crypto.createCipheriv.mockReturnValue(false);
+        createCipheriv.mockReturnValue(false);
 
-        crypto.pbkdf2Sync = jest.fn(() => {
-            return Buffer.from('0000000000000000');
-        });
+        pbkdf2Sync.mockReturnValueOnce(Buffer.from('0000000000000000'));
 
         keccak256.mockReturnValueOnce('0xmac');
 
@@ -542,11 +502,11 @@ describe('AccountTest', () => {
 
         expect(fromPrivate).toHaveBeenCalledWith('pk');
 
-        expect(crypto.randomBytes).toHaveBeenNthCalledWith(1, 32);
+        expect(randomBytes).toHaveBeenNthCalledWith(1, 32);
 
-        expect(crypto.randomBytes).toHaveBeenNthCalledWith(2, 16);
+        expect(randomBytes).toHaveBeenNthCalledWith(2, 16);
 
-        expect(crypto.pbkdf2Sync).toHaveBeenCalledWith(
+        expect(pbkdf2Sync).toHaveBeenCalledWith(
             Buffer.from('password'),
             Buffer.from('random'),
             262144,
@@ -554,7 +514,7 @@ describe('AccountTest', () => {
             'sha256'
         );
 
-        expect(crypto.createCipheriv).toHaveBeenCalledWith(
+        expect(createCipheriv).toHaveBeenCalledWith(
             'aes-128-ctr',
             Buffer.from('0000000000000000').slice(0, 16),
             Buffer.from('random')

--- a/packages/web3-providers/tests/src/providers/HttpProviderTest.js
+++ b/packages/web3-providers/tests/src/providers/HttpProviderTest.js
@@ -327,9 +327,7 @@ describe('HttpProviderTest', () => {
             xhrMock.onreadystatechange();
         }, 1);
 
-        await expect(httpProvider.sendPayload({id: '0x0'})).rejects.toThrow(
-            `HttpProvider ERROR: NOPE (code: 450)`
-        );
+        await expect(httpProvider.sendPayload({id: '0x0'})).rejects.toThrow('HttpProvider ERROR: NOPE (code: 450)');
 
         expect(providersModuleFactoryMock.createXMLHttpRequest).toHaveBeenCalledWith(
             httpProvider.host,

